### PR TITLE
Add reviews to slot page

### LINKED ISSUE #234: Show reviews on the website

### PR DESCRIPTION
**Pull request by jvyden**
Here's what it looks like on a level with mixed reviews:
image

Vita's review icons are a little smaller for some strange reason, but they still look alright:
image

- [x] Depends on #267

*Closes #234*

### CODE REVIEW

File: ProjectLighthouse/Pages/SlotPage.cshtml [hunk 1]
```diff
@@ -53,14 +57,84 @@
...
 
-@await Html.PartialAsync("Partials/CommentsPartial")
+            @foreach (Review review in Model.Reviews)
+            {
+                string faceHash = review.Thumb switch {
+                    -1 => review.Reviewer.BooHash,
+                    0 => review.Reviewer.MehHash,
+                    1 => review.Reviewer.YayHash,
+                    
+                    _ => throw new ArgumentOutOfRangeException(),
```
Comment (Slendy) [inline ProjectLighthouse/Pages/SlotPage.cshtml:90]: The /postReview endpoint doesn't clamp the Thumb value so a user can make their profile cause an error. This should either have a default value or the Thumb value needs to properly get limited in ReviewController.

File: ProjectLighthouse/Pages/SlotPage.cshtml.cs [hunk 2]
```diff
@@ -43,6 +45,19 @@ public async Task<IActionResult> OnGet([FromRoute] int id)
             this.Comments = new List<Comment>();
         }
 
+        if (this.ReviewsEnabled)
+        {
+            this.Reviews = await this.Database.Reviews.Include(r => r.Reviewer)
+                .OrderByDescending(r => r.Timestamp)
```
Comment (Slendy) [inline ProjectLighthouse/Pages/SlotPage.cshtml.cs:51]:
```suggestion
                .OrderByDescending(r => r.ThumbsUp - r.ThumbsDown)
                .ThenByDescending(r => r.Timestamp)
```
This should be ordered by either thumbsUp or the result of thumbs up minus thumbs down

**Review by Slendy — CHANGES_REQUESTED** [review]
Nice work. I only noticed a few small things that could be improved. I also think in the future we should add pages to reviews, comments, and photos on user and slot pages to conserve space.

File: ProjectLighthouse/Pages/SlotPage.cshtml (hunk 1 above)
Comment (jvyden) [reply ProjectLighthouse/Pages/SlotPage.cshtml:90]: Good to know. Are there any other places you can think of where ratings aren't clamped?

Comment (Slendy) [reply ProjectLighthouse/Pages/SlotPage.cshtml:90]: In my purge fixes branch I already fixed comment ratings not being clamped, but you could fix it here if you wanted

Comment (jvyden) [reply ProjectLighthouse/Pages/SlotPage.cshtml:90]: Hmm. If that's on a separate commit I could cherry-pick it into master. If not I can fix it in this MR. Up to you.

Comment (Slendy) [reply ProjectLighthouse/Pages/SlotPage.cshtml:90]: It'll be one big commit so I'll just do it in my PR.

Comment (jvyden) [reply ProjectLighthouse/Pages/SlotPage.cshtml:90]: Cool. I'll wait until then.

**Review by Slendy — APPROVED** [review]

**Pull request merged**